src/main/java/de/dennisguse/opentracks/services/GpsStatus.java

```diff
@@ -0,0 +1,217 @@
+
+    // If recording interval not set in preferences then it uses 10 seconds like signal lost threshold.
+    private static final int SIGNAL_LOST_THRESHOLD_DEFAULT = 10000;
+
+    // Threshold for accuracy.
+    private double signalBadThreshold;
+    // Threshold for time without points.
+    private int signalLostThreshold;
+
+    private GpsStatusValue gpsStatus = GpsStatusValue.GPS_NONE;
+    private GpsStatusListener service;
+    private Context context;
+    private Location lastLocation = null;
+    private Location lastValidLocation = null;
+
+    private class GpsStatusRunner implements Runnable {
+        private boolean stopped = false;
+
+        @Override
+        public void run() {
+            if (gpsStatus != null && !stopped) {
+                if (!stopped) {
+                    onLocationChanged(null);
+                    gpsStatusHandler.postDelayed(gpsStatusRunner, getIntervalThreshold());
+                }
+            }
+        }
+
+        public void stop() {
+            stopped = true;
+        }
+    }
+
+    private Handler gpsStatusHandler = null;
+    private GpsStatusRunner gpsStatusRunner = null;
+
+    public GpsStatus(Context context, GpsStatusListener service) {
+        this.service = service;
+        this.context = context;
+        signalBadThreshold = PreferencesUtils.getRecordingDistanceInterval(context);
+        signalLostThreshold = getRecordingIntervalInMillis(PreferencesUtils.getMinRecordingInterval(context));
+        gpsStatusHandler = new Handler();
+    }
+
+    /**
+     * The service that uses GpsStatus has to call this method to stop the Runnable if needed.
+     */
+    public void stop() {
+        if (gpsStatusRunner != null) {
+            gpsStatusRunner.stop();
+            gpsStatusRunner = null;
+        }
+    }
+
+    /**
+     * Method to change the bad threshold from outside.
+     *
+     * @param value New preference value to signalBadThreshold.
+     */
+    public void onRecordingDistanceChanged(int value) {
+        signalBadThreshold = value;
+    }
+
+    /**
+     * Method to change the lost threshold from outside.
+     *
+     * @param value New preference value to signalLostThreshold.
+     */
+    public void onMinRecordingIntervalChanged(int value) {
+        signalLostThreshold = getRecordingIntervalInMillis(value);
+    }
+
+    /**
+     * @param value the preference value for recording interval.
+     * @return the preference value in milliseconds or SIGNAL_LOST_THRESHOLD_DEFAULT if the preference value is a default one.
+     */
+    private int getRecordingIntervalInMillis(int value) {
+        String minDefault = context.getString(R.string.min_recording_interval_default);
+        String minAdaptAccuracy = context.getString(R.string.min_recording_interval_adapt_accuracy);
+        String minAdaptBatteryLife = context.getString(R.string.min_recording_interval_adapt_battery_life);
+        if (value != Integer.valueOf(minDefault) &&
```

Comment:
switch statement

Reply:
I rewrite it because values below 1 are special meaning, so it's enough to check if the value is greater than 0.